pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -184,7 +184,8 @@ func (e *EndpointController) getPodServiceMemberships(pod *api.Pod) (sets.String
 	for i := range services {
 		key, err := keyFunc(&services[i])
```

Comment:
Did you actually encounter this error? 
GetPodServices returns a list of services, so it's a very unlikely condition. Returning an error should force a requeue (I haven't checked, but i'd expect it to). Just returning one less service in the list will mean that service never gets synced, or doesn't get synced till next relist which could be O(minutes) at least.